### PR TITLE
Replace article structure with space-filler

### PR DESCRIPTION
**Pull request by gtrufitt**
## What does this change?

This replaces my 'are the last 5 elements paragraphs' rudimentary check with the super-powered spacefinder check via space-filler for the in-article email sign-up insertion.
## What is the value of this and can you measure success?

It's less likely to cause clashes and no longer breaks certain contracts.
## Request for comment

@crifmulholland on the rules, @jbreckmckye because you're spacefinderman

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/email/email-article.js [hunk 1]
```diff
@@ -101,34 +116,26 @@ define([
...
-                    $iframeEl[listConfig.insertMethod || 'appendTo']($insertEl && $insertEl.length > 0 ? $insertEl : $articleBody);
-                });
+                if (listConfig.insertMethod) {
+                    fastdom.write(function () {
+                        listConfig.insertMethod($iframeEl);
+                    });
+                } else {
+                    spaceFiller.fillSpace(getSpacefinderRules(), function (paras) {
+                        $iframeEl.insertBefore(paras[0]);
+                        emailInserted = true;
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/email/email-article.js]: Given that this is async, is there a risk that `init()` will be called in the interval before spaceFiller has had a chance to run your write function?

Comment (gtrufitt) [inline static/src/javascripts/projects/common/modules/email/email-article.js]: Are you referring to this `init`?

```
bean.on(iframe, 'load', function () {
    email.init(iframe);
});
```

If so, as it's inserting an iframe, that `load` event won't fire until it's inserted in the page and made the request so there shouldn't ever be a race condition there I don't think.

Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/email/email-article.js]: Fair enough. Just wanted to mention that `spaceFiller` returns a promise, if you need it.

**Comment by crifmulholland** [comment]
The rules look good to me. The only other thing we need is tracking when it's shown - can we please fire an event when rendered onto the page?

File: static/src/javascripts/projects/common/modules/email/email-article.js (hunk 1 above)
Comment (gtrufitt) [inline static/src/javascripts/projects/common/modules/email/email-article.js]: Aye, thanks, saw that but didn't need it in this case, cheers!

**Comment by jbreckmckye** [comment]
:+1:

**Pull request merged**